src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -140,38 +147,94 @@ internal void Freeze()
             _orderedSynthesizedMethods = _synthesizedMethods.OrderBy(kvp => kvp.Key).Select(kvp => kvp.Value).AsImmutable();
 
             // Sort proxy types.
-            _orderedProxyTypes = _proxyTypes.OrderBy(kvp => kvp.Key).Select(kvp => kvp.Value).AsImmutable();
+            _orderedProxyTypes = _proxyTypes.OrderBy(kvp => kvp.Key.Size).Select(kvp => kvp.Value).AsImmutable();
         }
 
         private bool IsFrozen => _frozen != 0;
 
-        internal Cci.IFieldReference CreateDataField(ImmutableArray<byte> data)
+        /// <summary>
+        /// Gets a field that can be used to cache an array allocated to store data from a corresponding <see cref="CreateDataField"/> call.
+        /// </summary>
+        /// <param name="data">The data that will be used to initialize the field.</param>
+        /// <param name="arrayType">The type of the field, e.g. int[].</param>
+        /// <param name="emitContext">The emit context to use with the array type to extract its element type.</param>
+        /// <returns>The field to use to cache an array for this data and alignment.</returns>
+        internal Cci.IFieldReference CreateArrayCachingField(ImmutableArray<byte> data, Cci.IArrayTypeReference arrayType, EmitContext emitContext)
         {
             Debug.Assert(!IsFrozen);
-            Cci.ITypeReference type = _proxyTypes.GetOrAdd((uint)data.Length, GetStorageStruct);
-            return _mappedFields.GetOrAdd(data, data0 =>
+
+            // Get the type code for the array's element type.  It'll be used part of the
```

Comment:
>It'll be used part of the [](http://example.com/codeflow?start=64&length=25)

It looks like the sentence is not finished

Reply:
I have no idea what I intended to write as that second sentence :) I'm just going to delete it.